api/envoy/type/http/v3/header.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package envoy.type.http.v3;
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.type.http.v3";
+option java_outer_classname = "HeaderProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/type/http/v3;httpv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Http header API]
+
+// Header defines an API for obtaining HTTP Header.
+message Header {
```

Comment:
This seems like a very odd type.  It's called "Header", but it includes both a header name and a path.

This doesn't seem like the sort of thing that will ever be used in any other context.  I think these fields should just be inlined into the `HeaderBasedSessionState` message.

Reply:
I agree. The idea was to keep it consistent with the `CookieBasedSessionState` [message](https://github.com/envoyproxy/envoy/blob/0d0596315a40f0da502af1a9a885d6def0e5e02e/api/envoy/extensions/http/stateful_session/cookie/v3/cookie.proto#L39).
I can inline this and the updated config would look like this ->
```
http_filters:
 - name: envoy.filters.http.stateful_session
   typed_config:
     "@type": type.googleapis.com/envoy.extensions.filters.http.stateful_session.v3.StatefulSession
     session_state:
       name: envoy.http.stateful_session.header
       typed_config:
         "@type": type.googleapis.com/envoy.extensions.http.stateful_session.header.v3.HeaderBasedSessionState
         name: x-sticky-session-custom-header
         path: /session/v1/
```
Which is alright I guess 🤔 
@wbpcode wdyt?

Reply:
I think this should be just a special config for this new extension. 🤔 So, we shouldn't add this `Header` to the `type.http.v3`.

And is the path necessary for header based stateful session?

Reply:
> And is the path necessary for header based stateful session?

Yes, because:
- It keeps the client simple. It can just pass the header every time.
- More importantly, Envoy knows when it is expected to send session info.
    If not path information is present with Envoy then it won't be able to differentiate between:
          1. client not sending the header because it doesn't want stickiness for this call vs 
          2. client not sending the session header because this is the first call.

Reply:
> * More importantly, Envoy knows when it is expected to send session info.
>   If not path information is present with Envoy then it won't be able to differentiate between:
>   1. client not sending the header because it doesn't want stickiness for this call vs
>   2. client not sending the session header because this is the first call.

I'm confused, this doesn't seem to be related with `path` here.

Reply:
For `StatefulSessionPerRoute`, the path isn't needed. Since that is already [per route](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/stateful_session/v3/stateful_session.proto#extensions-filters-http-stateful-session-v3-statefulsessionperroute).
But `StatefulSession` [config](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/stateful_session/v3/stateful_session.proto#extensions-filters-http-stateful-session-v3-statefulsession) is much larger in scope, so we need a way to convey when stickiness is expected.

Reply:
Done.

Reply:
Done. Removed the `path`